bodhi/tests/server/services/test_updates.py

```diff
@@ -932,6 +932,83 @@ def test_provenpackager_request_update_with_none_ci(self, publish, *args):
             res.json_body['errors'][0]['description'],
             config.get('not_yet_tested_msg'))
 
+    @mock.patch(**mock_taskotron_results)
+    @mock.patch(**mock_valid_requirements)
+    @mock.patch('bodhi.server.notifications.publish')
+    def test_provenpackager_request_update_with_gw_satisfied(self, publish, *args):
+        """Ensure provenpackagers can request changes for any update which
+        satisfies Greenwave policies"""
+        nvr = u'bodhi-2.1-1.fc17'
+        user = User(name=u'bob')
+        self.db.add(user)
+        group = self.db.query(Group).filter_by(name=u'provenpackager').one()
+        user.groups.append(group)
+        settings = self.app_settings.copy()
+        settings['greenwave_enabled'] = True
+        self.db.commit()
+
+        app = TestApp(main({}, testing=u'bob', session=self.db, **settings))
```

Comment:
It seems like if I change the settings in self.app, it will interfere with other tests in the class.  Maybe I should set it at start and then set it back at the end of the test?